spec/models/game_spec.rb

```diff
@@ -148,11 +148,46 @@
     end
 
     it 'returns false if king is currently in check' do
-      game = FactoryGirl.create(:game)
+      game = FactoryGirl.create(:game, move_number: 10)
       white_king = FactoryGirl.create(:king, is_white: true, game: game, x_position: 4, y_position: 7)
-      FactoryGirl.create(:rook, is_white: false, game: game, x_position: 4, y_position: 5)
+      FactoryGirl.create(:rook, is_white: false, game: game, x_position: 4, y_position: 5, game_move_number: 10)
       FactoryGirl.create(:king, is_white: false, game: game, x_position: 5, y_position: 2)
       expect(game.stalemate?(white_king.is_white)).to eq(false)
     end
   end
+
+  describe '#checkmate?' do
+    it 'should return true if king is in checkmate' do
+      game = FactoryGirl.create(:game, move_number: 15)
+      FactoryGirl.create(:king, is_white: false, game: game, x_position: 8, y_position: 5)
+      FactoryGirl.create(:king, is_white: true, game: game, x_position: 6, y_position: 5)
+      FactoryGirl.create(:rook, is_white: true, game: game, x_position: 8, y_position: 1, game_move_number: 15)
+      expect(game.checkmate?).to eq(true)
+    end
+
+    it 'should return false if king can capture attacking piece' do
+      game = FactoryGirl.create(:game)
+      FactoryGirl.create(:king, is_white: true, game: game, x_position: 6, y_position: 3)
+      FactoryGirl.create(:king, is_white: false, game: game, x_position: 5, y_position: 8)
+      FactoryGirl.create(:rook, is_white: false, game: game, x_position: 5, y_position: 3)
+      expect(game.checkmate?).to eq(false)
+    end
```

Comment:
I would like to suggest changing this by adding black rook to (6, 4), so that white king has to capture an attacker to get out of checkmate.

Reply:
Sounds good to me!

Reply:
I noticed that in the test 'should return false if king can capture attacking piece there is no second black rook threatening the white king from (6, 4), which that white king can capture.